server/tso/allocator_manager.go

```diff
@@ -1081,6 +1118,35 @@ func (am *AllocatorManager) setGRPCConn(newConn *grpc.ClientConn, addr string) {
 	am.localAllocatorConn.clientConns[addr] = newConn
 }
 
+func (am *AllocatorManager) transferLocalAllocator(dcLocation string, targetServerID uint64) error {
+	serverID := targetServerID
+	nextLeaderKey := path.Join(am.rootPath, dcLocation, "next-leader")
+	// Grant a etcd lease with checkStep * 1.5
+	nextLeaderLease := clientv3.NewLease(am.member.Client())
+	ctx, cancel := context.WithTimeout(am.member.Client().Ctx(), etcdutil.DefaultRequestTimeout)
+	leaseResp, err := nextLeaderLease.Grant(ctx, int64(checkStep.Seconds()*1.5))
+	cancel()
+	if err != nil {
+		err = errs.ErrEtcdGrantLease.Wrap(err).GenWithStackByCause()
+		log.Error("failed to grant the lease of the next leader id key", errs.ZapError(err))
+		return err
+	}
+	resp, err := kv.NewSlowLogTxn(am.member.Client()).
+		If(clientv3.Compare(clientv3.CreateRevision(nextLeaderKey), "=", 0)).
+		Then(clientv3.OpPut(nextLeaderKey, fmt.Sprint(serverID), clientv3.WithLease(leaseResp.ID))).
+		Commit()
+	if err != nil {
+		err = errs.ErrEtcdTxn.Wrap(err).GenWithStackByCause()
+		log.Error("failed to write next leader id into etcd", errs.ZapError(err))
+		return err
+	}
+	if !resp.Succeeded {
+		log.Warn("write next leader id into etcd unsuccessfully", zap.String("dc-location", dcLocation))
+		return errs.ErrEtcdKVPut.GenWithStack("write next leader id into etcd unsuccessfully")
+	}
+	return nil
```

Comment:
These codes are duplicated with those in `PriorityChecker`. We should write a function to do this.